src/pyodide/internal/python.js

```diff
@@ -2,6 +2,23 @@ import { parseTarInfo } from "pyodide-internal:tar";
 import { createTarFS } from "pyodide-internal:tarfs";
 import { createMetadataFS } from "pyodide-internal:metadatafs";
 import { default as ArtifactBundler } from "pyodide-internal:artifacts";
+import { default as LOCKFILE } from "pyodide-internal:generated/pyodide-lock.json";
+
+const canonicalizeNameRegex = /[-_.]+/g;
+
+/**
+ * Normalize a package name. Port of Python's packaging.utils.canonicalize_name.
+ * @param name The package name to normalize.
+ * @returns The normalized package name.
+ * @private
+ */
+export function canonicalizePackageName(name) {
+  return name.replace(canonicalizeNameRegex, "-").toLowerCase();
+}
+
+const STDLIB_PACKAGES = Object.values(LOCKFILE.packages)
+  .filter(({install_dir}) => install_dir === "stdlib")
+  .map(({name}) => canonicalizePackageName(name));
```

Comment:
The name field should already be normalized. If this actually is necessary for some reason, we should also open an upstream bug.

Reply:
It's necessary because the name is normalized in the key but not in the name field for some reason.

Reply:
I can rewrite it to use the name in the object key

Reply:
Interesting. Well that's fine then, maybe just put a comment saying that and leave as-is. And open a bug on Pyodide asking if this is intentional.